### PR TITLE
Add the Windows App SDK to the packs

### PR DESCRIPTION
**Pull request by mattleibow**
### Description of Change

Speed up builds and stuff.

### CODE REVIEW

File: eng/cake/dotnet.cake [hunk 1]
```diff
@@ -205,6 +208,44 @@ Task("dotnet-pack")
...
+            CopyFiles($"{tempDir}/**/" + id + ".nupkg", destDir, false);
+            CleanDirectories(tempDir);
+        }
+
+        Download("Microsoft.Maui.Graphics", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json");
+        Download("Microsoft.Maui.Graphics.Win2D.WinUI.Desktop", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.WindowsAppSDK", "MicrosoftWindowsAppSDKPackageVersion", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.Windows.SDK.BuildTools", "MicrosoftWindowsSDKBuildToolsPackageVersion", "https://api.nuget.org/v3/index.json");
+
+        Zip(destDir, $"{destDir}.zip");
```
Comment (pjcollins) [inline eng/cake/dotnet.cake:240]: Do we need to zip/unzip these? Should we maybe just upload them as a separate `library-packs` azure artifact to save a couple steps?

Comment (pjcollins) [reply eng/cake/dotnet.cake:240]: Once we add a new artifact for this, we should add another download step to the top of the sdk-insertion.yml job:

```yaml
  - task: DownloadPipelineArtifact@2
    inputs:
      artifactName: library-packs
      downloadPath: $(Build.StagingDirectory)\nuget
      patterns: |
        **/Microsoft.Maui.Graphics.*.nupkg
```

File: eng/cake/dotnet.cake [hunk 2]
```diff
@@ -205,6 +208,42 @@ Task("dotnet-pack")
...
+            });
+
+            CopyFiles($"{tempDir}/**/" + id + ".nupkg", destDir, false);
+            CleanDirectories(tempDir);
+        }
+
+        Download("Microsoft.Maui.Graphics", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json");
+        Download("Microsoft.Maui.Graphics.Win2D.WinUI.Desktop", "MicrosoftMauiGraphicsVersion", "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.WindowsAppSDK", "MicrosoftWindowsAppSDKPackageVersion", "https://api.nuget.org/v3/index.json");
+        Download("Microsoft.Windows.SDK.BuildTools", "MicrosoftWindowsSDKBuildToolsPackageVersion", "https://api.nuget.org/v3/index.json");
```
Comment (pjcollins) [inline eng/cake/dotnet.cake:239]: It looks like the msi generation tooling is expecting these files to have the package version in the file name:
```
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Maui.Graphics.6.0.300-rc.1.1198.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Maui.Graphics.Win2D.WinUI.Desktop.6.0.300-rc.1.1198.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.WindowsAppSDK.1.0.1.nupkg
Workload pack package does not exist: D:\a\_work\1\a\nugets\signed\Microsoft.Windows.SDK.BuildTools.10.0.22000.194.nupkg
```

We'll need to add the version to file name on download or when copying the file.

Comment (mattleibow) [reply eng/cake/dotnet.cake:239]: Fixed this and the msi seem to be in the artifacts.

**Comment by mattleibow** [comment]
This was a build that seemed to do what we want: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=5953982&view=results

**Review by pjcollins — APPROVED** [review]
Looks good to me, assuming it works as expected to improve certain WinUI build situations.

**Pull request merged**